### DISCUSSION TITLE
Add instruction in how to make changes to a PR

### DISCUSSION THREAD

**Discussion by moonbe77** [Syknapse/Contribute-To-This-Project#959] [Ideas]
Hey, I wonder if we should add instructions in how is the process of making changes to a PR after someone does the review.
I notice there is a lot of PR that had been reviewed but no changes are made to the PR to be able to merge.

Is this something that could help?

cheers

**Comment by lax0248917** [comment]
This sounds like a good idea, we either good add some instructions into the readme, or we could make sure we are expressing the process in our code review comments. It helps to point out that it even states in the PR where to commit your changes so that they are reflected in the PR once they are pushed.

**Reply by ghost** [reply]
great comment!

**Comment by Syknapse** [comment]
I think it's a great idea to have as much detailed instructions as possible to what the process is after submitting your PR.
Maybe the best way to do it is have another markdown document especially for that purpose and link to it clearly from the readme. This way we keep the readme exclusively dedicated to the contribution and not have too much information in one place

**Reply by 25Ericcheong** [reply]
Would adding a `pull_request_guide.md` and linking it in the `README.md` file be worth it? Could add the link in the `Next steps` section perhaps?

**Comment by roramigator** [comment]
I'd also suggest to have a limit on how long a PR can be open after its review; there's a few PRs open that are 7 months old, could this be something to consider?

**Comment by Syknapse** [comment]
Do we have any volunteers to take on this really good idea?

**Reply by Preet-Sojitra** [reply]
> Do we have any volunteers to take on this really good idea?

I think older PR, in which they were told to make change, and haven't made the changes those PR's should be closed.

**Comment by Veercodeprog** [comment]
Fork the repository: Navigate to the repository you want to contribute to and click the "Fork" button in the top-right corner of the page. This will create a copy of the repository in your own GitHub account.

Clone the repository: On your own GitHub repository page, click the green "Code" button and copy the URL. Then, in your terminal, navigate to the directory where you want to store the repository and run the git clone command followed by the repository URL.

Create a new branch: Once you've cloned the repository, create a new branch for your changes. You can do this by running the git checkout -b <branch-name> command, where <branch-name> is the name of your new branch.

Make changes: Make any desired changes to the code or files in your new branch.

Commit changes: Once you've made your changes, use git add to add any modified or new files to the staging area, and then use git commit to commit the changes with a descriptive message.

Push changes: Push your changes to your forked repository by running git push origin <branch-name>. This will upload your changes to your forked repository.

Create a pull request: Go to the original repository where you want to contribute and click the "New pull request" button. GitHub will automatically detect your changes and ask you to confirm which branch you want to merge into the original repository's branch. Review the changes and add any additional information, such as a description or screenshot, to the pull request. Then click the "Create pull request" button to submit it.

Review and merge: The repository owner or maintainer will review your pull request and may ask for additional changes or clarification. Once the pull request is approved, it can be merged into the original repository by the owner or a designated team member.
